src/electrum_wallet_file.rs

```diff
@@ -0,0 +1,438 @@
+use crate::{ElectrumExtendedKey, ElectrumExtendedPrivKey, ElectrumExtendedPubKey};
+use bitcoin::util::bip32::{ExtendedPrivKey, ExtendedPubKey};
+use regex::Regex;
+use serde::{de, ser::SerializeMap, Deserialize, Deserializer, Serialize, Serializer};
+use std::{fmt, path::Path, str::FromStr, string::ToString};
+
+/// Representation of an electrum wallet file
+#[derive(Clone, Debug, PartialEq)]
+pub struct ElectrumWalletFile {
+    pub addresses: Addresses,
+    pub wallet_type: WalletType,
+    pub keystores: Vec<Keystore>,
+}
+
+impl ElectrumWalletFile {
+    /// Parse an electrum wallet file
+    pub fn from_file(wallet_file: &Path) -> Result<Self, String> {
+        let file = std::fs::File::open(wallet_file).map_err(|e| e.to_string())?;
+        let wallet = serde_json::from_reader(file).map_err(|e| e.to_string())?;
+        Ok(wallet)
+    }
+
+    /// Write to an electrum wallet file
+    pub fn to_file(&self, wallet_file: &Path) -> Result<(), String> {
+        let file = std::fs::File::create(wallet_file).map_err(|e| e.to_string())?;
+        serde_json::to_writer_pretty(file, self).map_err(|e| e.to_string())
+    }
+
+    /// Convert from an output descriptor. Only the external descriptor is needed, the change descriptor is implied.
+    pub fn from_descriptor(desc: &str) -> Result<Self, String> {
+        if desc.contains("sortedmulti") {
+            ElectrumWalletFile::from_descriptor_multisig(desc)
+        } else {
+            ElectrumWalletFile::from_descriptor_singlesig(desc)
+        }
+    }
+
+    fn from_descriptor_singlesig(desc: &str) -> Result<Self, String> {
+        let re =
+            Regex::new(r#"(pkh|sh\(wpkh|sh\(wsh|wpkh|wsh)\((([tx]p(ub|rv)[0-9A-Za-z]+)/0/\*)\)+"#)
```

Comment:
Yes, let's explore that in the future.

Reply:
Also for things to exlore in the future, I thought about:
* returning a simple struct with an internal and an  external field instead of a Vec with two fields in various functions. 
* a typedef for the common error type of the lib. It can still define it as String, but using the Error type would look better in the function signatures.

Reply:
Yeah, I thought about the needs for the crate Error type myself yesterday. And also the other one is a good point

Reply:
Added issues...